### PR TITLE
4668290: unclear spec for Polygon.bounds field

### PR DESCRIPTION
**Pull request by SWinxy**
New documentation replaces the `This value can be null.`:
```
When created, {@link #invalidate() invalidated}, or {@link #reset() reset},
this becomes {@code null}. To get out of the null state,
{@link #getBounds()} called with {@link #npoints} being greater than
{@code 0} will transfer into a non-null {@link Rectangle}.
```

I think I've got the reason for why it can be `null` correct, but you never know.

In javax.swing.text.html.Map, I've replaced setting the field to null with the equivalent #invalidate(). This may mean we can make the field private in the future.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [ ] Change requires CSR request [JDK-8298864](https://bugs.openjdk.org/browse/JDK-8298864) to be approved
- [x] Commit message must refer to an issue

### Issues
 * [JDK-4668290](https://bugs.openjdk.org/browse/JDK-4668290): unclear spec for Polygon.bounds field
 * [JDK-8298864](https://bugs.openjdk.org/browse/JDK-8298864): unclear spec for Polygon.bounds field (**CSR**)

### Reviewers
 * [Alexey Ivanov](https://openjdk.org/census#aivanov) (@aivanov-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10417/head:pull/10417` \
`$ git checkout pull/10417`

Update a local copy of the PR: \
`$ git checkout pull/10417` \
`$ git pull https://git.openjdk.org/jdk pull/10417/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10417`

View PR using the GUI difftool: \
`$ git pr show -t 10417`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10417.diff">https://git.openjdk.org/jdk/pull/10417.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/share/classes/java/awt/Polygon.java [hunk 1]
```diff
@@ -103,7 +103,11 @@ public class Polygon implements Shape, java.io.Serializable {
 
     /**
      * The bounds of this {@code Polygon}.
-     * This value can be null.
+     * <p>
+     * When created, {@link #invalidate() invalidated}, or {@link #reset() reset},
+     * this becomes {@code null}. To get out of the null state,
+     * {@link #getBounds()} called with {@link #npoints} being greater than
+     * {@code 0} will transfer into a non-null {@link Rectangle}.
      *
```
Comment (prrace) [inline src/java.desktop/share/classes/java/awt/Polygon.java:111]: Another dubious piece of design. This should never have been exposed in the API.
What we need to say here is
/*
 * The value of this field is updated by the implementation which may set it to null and will
 *   re-evaluate it as needed. Application sub-classes should never access this field.
 */
And leave it at that.

File: src/java.desktop/share/classes/java/awt/Polygon.java [hunk 2]
```diff
@@ -103,7 +103,10 @@ public class Polygon implements Shape, java.io.Serializable {
 
     /**
      * The bounds of this {@code Polygon}.
-     * This value can be null.
+     * <p>
+     * The value of this field is updated by the implementation which
+     * may set it to null and will reevaluate it as needed. Application
```
Comment (aivanov-jdk) [inline src/java.desktop/share/classes/java/awt/Polygon.java:108]: I'd use `{@code null}` syntax.

**Review by aivanov-jdk — APPROVED** [review]

File: src/java.desktop/share/classes/java/awt/Polygon.java (hunk 1 above)
Comment (aivanov-jdk) [reply src/java.desktop/share/classes/java/awt/Polygon.java:110]: I noticed one thing: Phil suggested “re-evaluate”, you used “reevaluate” without the hyphen.

I tend to use the hyphened version; at the same time, articles in the internet suggest the hyphen is unneeded in such words.

I'm fine with either.

**Review by prrace — APPROVED** [review]
Need the CSR

**Review by aivanov-jdk — APPROVED** [review]

**Pull request closed without merging**